File: packages/ogdf.js/src/layouts/energybased/StressMinimization.js
import { PARAMETER_TYPE } from '../../utils/parameter-type'

export default {
    terminationCriterion: {
        type: PARAMETER_TYPE.CATEGORICAL,
        range: ['None', 'PositionDifference', 'Stress'],
        default: 'None'
    },
    fixXCoords: {
        type: PARAMETER_TYPE.BOOL,
        range: [true, false],
        default: false
    },
    fixYCoords: {
        type: PARAMETER_TYPE.BOOL,
        range: [true, false],
        default: false
    },
    fixZCoords: {
        type: PARAMETER_TYPE.BOOL,
        range: [true, false],
        default: false
    },
    hasInitialLayout: {
        type: PARAMETER_TYPE.BOOL,
        range: [true, false],
        default: false
    },
    layoutComponentsSeparately: {
        type: PARAMETER_TYPE.BOOL,
        range: [true, false],
        default: false
    },
    edgeCosts: {
        type: PARAMETER_TYPE.DOUBLE,
        range: [0, Infinity],
        default: 100
    },
    numberOfIterations: {
        type: PARAMETER_TYPE.INT,
        range: [0, Infinity],
        default: 200
    },
    useEdgeCostsAttribute: {
        type: PARAMETER_TYPE.BOOL,
        range: [true, false],
        default: false
    }
}

File: packages/ogdf.js/src/entry/energybased/StressMinimization.cpp
#include <ogdf/energybased/StressMinimization.h>
#include "../main.h"

EM_PORT_API(StressMinimization *)
LayoutModule_StressMinimization(int terminationCriterion, bool fixXCoords, bool fixYCoords, bool fixZCoords, bool hasInitialLayout, bool layoutComponentsSeparately, double edgeCosts, int numberOfIterations, bool useEdgeCostsAttribute)
{
    //LayoutModule
	StressMinimization *model = new StressMinimization();

	model->convergenceCriterion(static_cast<StressMinimization::TerminationCriterion>(terminationCriterion));
	model->fixXCoordinates(fixXCoords);
	model->fixYCoordinates(fixYCoords);
	model->fixZCoordinates(fixZCoords);
	model->hasInitialLayout(hasInitialLayout);
	model->layoutComponentsSeparately(layoutComponentsSeparately);
	model->setEdgeCosts(edgeCosts);
	model->setIterations(numberOfIterations);
	model->useEdgeCostsAttribute(useEdgeCostsAttribute);

    return model;
}